Cargo.toml

```diff
@@ -15,12 +15,18 @@ num-traits = { version = "0.2", default-features = false }
 ecdsa = { version = "0.16", default-features = false, features = ["arithmetic", "digest", "hazmat"] }
 sha2 = { version = "0.10", default-features = false }
 
+# Optional dependencies
+serdect = { version = "0.3.0-rc.0", optional = true, default-features = false }
```

Comment:
Why do we need an explicit `serdect` dependency?

Reply:
Not sure! Tbh, I opened this draft with the bare minimum copy&paste to make the issue I was facing in synedrion go away.

Reply:
I think we need it (or serde itself) for the traits. It doesn't seem like primeorder re-exports it.

`#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]`

Reply:
`serde`, certainly, but we don't use anything from `serdect`

Reply:
We also need the "derive" feature, so there's that.

Reply:
Not sure I understand, it can be enabled in `serde` itself, can't it?

Reply:
We can of course use `serde` directly (or even just `serde_derive`), but I thought there was a point in using `serdect` across all the RustCrypto crates, and that we wanted to be as compatible as possible with that ecosystem of crates.